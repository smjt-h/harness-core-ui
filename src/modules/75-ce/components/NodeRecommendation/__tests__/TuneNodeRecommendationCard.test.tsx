/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import { TuneRecommendationCard } from '../TuneNodeRecommendationCard'
import type { IState } from '../constants'

const mockState = {
  maxCpu: 44,
  maxMemory: 100,
  sumCpu: 30,
  sumMem: 120,
  minNodes: 10,
  includeTypes: [],
  includeSeries: [],
  excludeTypes: [],
  excludeSeries: []
}

const mockState2 = {
  maxCpu: 44,
  maxMemory: 100,
  sumCpu: 30,
  sumMem: 120,
  minNodes: 10,
  includeTypes: ['type1'],
  includeSeries: [],
  excludeTypes: ['type2'],
  excludeSeries: []
}

jest.mock('framework/strings', () => ({
  ...(jest.requireActual('framework/strings') as any),
  useStrings: jest.fn().mockReturnValue({
    getString: jest.fn().mockImplementation(val => val)
  })
}))

describe('test cases for tune recommendation card', () => {
  test('should be able to render tune recommendation card hidden', async () => {
    const { container } = render(
      <TestWrapper>
        <TuneRecommendationCard
          loading={false}
          cardVisible={false}
          toggleCardVisible={jest.fn()}
          buffer={0}
          dispatch={jest.fn()}
          setBuffer={jest.fn()}
          state={mockState}
          initialState={mockState as IState}
          showInstanceFamiliesModal={jest.fn()}
          updateRecommendationDetails={jest.fn()}
          updatedState={mockState as IState}
        />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })

  test('should be able to render tune recommendation card', async () => {
    const { container } = render(
      <TestWrapper>
        <TuneRecommendationCard
          loading={true}
          cardVisible={true}
          toggleCardVisible={jest.fn()}
          buffer={0}
          dispatch={jest.fn()}
          setBuffer={jest.fn()}
          state={mockState}
          initialState={mockState as IState}
          showInstanceFamiliesModal={jest.fn()}
          updateRecommendationDetails={jest.fn()}
          updatedState={mockState as IState}
        />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })

  test('should be able to render tune recommendation card when there are instance families', async () => {
    const { container } = render(
      <TestWrapper>
        <TuneRecommendationCard
          loading={true}
          cardVisible={true}
          toggleCardVisible={jest.fn()}
          buffer={0}
          dispatch={jest.fn()}
          setBuffer={jest.fn()}
          state={mockState2}
          initialState={mockState as IState}
          showInstanceFamiliesModal={jest.fn()}
          updateRecommendationDetails={jest.fn()}
          updatedState={mockState2 as IState}
        />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })
})
