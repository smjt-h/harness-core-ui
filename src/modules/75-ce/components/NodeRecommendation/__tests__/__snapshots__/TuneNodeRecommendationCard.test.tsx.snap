// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test cases for tune recommendation card should be able to render tune recommendation card 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main spaceBetween StyledProps--padding-medium StyledProps--flex"
  >
    <p
      class="StyledProps--font StyledProps--main pointer StyledProps--font-variation-h6 StyledProps--color StyledProps--color-grey800"
    >
      ce.nodeRecommendation.setInstancePreferences
    </p>
    <span
      class="bp3-icon bp3-icon-caret-up StyledProps--main pointer StyledProps--color StyledProps--color-primary7"
      icon="caret-up"
    >
      <svg
        data-icon="caret-up"
        height="16"
        viewBox="0 0 16 16"
        width="16"
      >
        <desc>
          caret-up
        </desc>
        <path
          d="M11.87 9.17s.01 0 0 0l-3.5-4C8.28 5.07 8.15 5 8 5s-.28.07-.37.17l-3.5 4a.495.495 0 00.37.83h7a.495.495 0 00.37-.83z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
  </div>
  <div
    class="StyledProps--font StyledProps--main preferences StyledProps--padding-medium StyledProps--background-primary1"
  >
    <div
      class="Layout--vertical Layout--layout-spacing-medium StyledProps--main"
    >
      <div
        class="StyledProps--font StyledProps--main"
      >
        <div
          class="Text--withDocsTooltip"
          data-tooltip-id="prefResourceNeeds"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-small-semi"
          >
            ce.nodeRecommendation.prefResourceNeeds
          </p>
        </div>
        <div
          class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main spaceBetween StyledProps--flex StyledProps--margin-top-small"
        >
          <div
            class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
          >
            <div
              class="StyledProps--font StyledProps--main"
            >
              <div
                class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
              >
                <div
                  class="StyledProps--font StyledProps--main spaceBetween StyledProps--flex"
                >
                  <span
                    class="StyledProps--font StyledProps--main StyledProps--padding-right-small StyledProps--inline StyledProps--color StyledProps--color-grey500 StyledProps--font-variation-small-semi"
                  >
                    ce.nodeRecommendation.cpus
                  </span>
                  <div
                    class="TextInput--main input"
                  >
                    <div
                      class="bp3-input-group"
                    >
                      <input
                        class="bp3-input"
                        data-testid="sumCpu-input"
                        type="text"
                        value="30"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="StyledProps--font StyledProps--main spaceBetween StyledProps--flex"
                >
                  <span
                    class="StyledProps--font StyledProps--main StyledProps--padding-right-small StyledProps--inline StyledProps--color StyledProps--color-grey500 StyledProps--font-variation-small-semi"
                  >
                    ce.nodeRecommendation.mem
                  </span>
                  <div
                    class="TextInput--main input"
                  >
                    <div
                      class="bp3-input-group"
                    >
                      <input
                        class="bp3-input"
                        data-testid="sumMem-input"
                        type="text"
                        value="120"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-center"
            >
              <div
                class="StyledProps--font StyledProps--main"
                style="border-width: 1px; border-style: solid; border-left: 0; width: 14px; height: 60px;"
              />
            </div>
            <div
              class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-center StyledProps--margin-right-small"
            >
              <span
                class="bp3-icon bp3-icon-plus StyledProps--main StyledProps--color StyledProps--color-grey700"
                icon="plus"
              >
                <svg
                  data-icon="plus"
                  height="12"
                  viewBox="0 0 16 16"
                  width="12"
                >
                  <desc>
                    plus
                  </desc>
                  <path
                    d="M13 7H9V3c0-.55-.45-1-1-1s-1 .45-1 1v4H3c-.55 0-1 .45-1 1s.45 1 1 1h4v4c0 .55.45 1 1 1s1-.45 1-1V9h4c.55 0 1-.45 1-1s-.45-1-1-1z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </div>
          </div>
          <div
            class="StyledProps--font StyledProps--main"
            style="width: 60%;"
          >
            <div
              class="Layout--vertical StyledProps--main"
            >
              <div
                class="Layout--horizontal StyledProps--main"
                style="justify-content: space-between;"
              >
                <div
                  class="Text--withDocsTooltip"
                  data-tooltip-id="buffer"
                >
                  <p
                    class="StyledProps--font StyledProps--main StyledProps--font-variation-small-semi"
                    style="display: flex; align-items: center;"
                  >
                    ce.nodeRecommendation.buffer
                    <span
                      class="bp3-icon StyledProps--main Text--icon StyledProps--padding-left-xsmall StyledProps--color StyledProps--color-grey800"
                      data-icon="info"
                    >
                      <svg
                        height="12"
                        viewBox="0 0 12 12"
                        width="12"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M0 6a6 6 0 1012 0A6 6 0 000 6zm.75 0a5.25 5.25 0 1010.5 0A5.25 5.25 0 00.75 6zm4.206 3.104v.646h2.682v-.646l-.836-.144V4.203H4.956v.65l.836.144V8.96l-.836.144zm1.846-7.352v1.03h-1.01v-1.03h1.01z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                  </p>
                </div>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-variation-small-semi StyledProps--color StyledProps--color-grey400"
                >
                  0%
                </p>
              </div>
              <div
                class="bp3-slider bp3-slider-unlabeled bufferSlider"
              >
                <div
                  class="bp3-slider-track"
                >
                  <div
                    class="bp3-slider-progress bp3-intent-primary"
                    style="left: 0.00%; right: 100.00%; top: 0px;"
                  />
                  <div
                    class="bp3-slider-progress"
                    style="left: 0.00%; right: 100.00%; top: 0px;"
                  />
                  <div
                    class="bp3-slider-progress"
                    style="left: 0.00%; right: 0.00%; top: 0px;"
                  />
                </div>
                <div
                  class="bp3-slider-axis"
                />
                <span
                  class="bp3-slider-handle"
                  style="left: calc(0.00% - 0px);"
                  tabindex="0"
                />
              </div>
              <p
                class="StyledProps--font StyledProps--main StyledProps--font-variation-small StyledProps--margin-top-xsmall"
              >
                ce.nodeRecommendation.cpuAndMemeValWithBuffer
              </p>
            </div>
          </div>
        </div>
        <div
          class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main StyledProps--padding-top-medium"
        >
          <div
            class="StyledProps--font StyledProps--main"
          >
            <div
              class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
            >
              <div
                class="Text--withDocsTooltip"
                data-tooltip-id="largestWorkloadReq"
              >
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-variation-small-semi"
                >
                  ce.nodeRecommendation.largestWorkloadReq
                </p>
              </div>
              <div
                class="StyledProps--font StyledProps--main spaceBetween StyledProps--flex"
                style="width: 75%;"
              >
                <span
                  class="StyledProps--font StyledProps--main StyledProps--inline StyledProps--color StyledProps--color-grey500 StyledProps--font-variation-small-semi"
                >
                  ce.nodeRecommendation.cpus
                </span>
                <div
                  class="TextInput--main input"
                >
                  <div
                    class="bp3-input-group"
                  >
                    <input
                      class="bp3-input"
                      data-testid="maxCpu-input"
                      type="text"
                      value="44"
                    />
                  </div>
                </div>
              </div>
              <div
                class="StyledProps--font StyledProps--main spaceBetween StyledProps--flex"
                style="width: 75%;"
              >
                <span
                  class="StyledProps--font StyledProps--main StyledProps--inline StyledProps--color StyledProps--color-grey500 StyledProps--font-variation-small-semi"
                >
                  ce.nodeRecommendation.mem
                </span>
                <div
                  class="TextInput--main input"
                >
                  <div
                    class="bp3-input-group"
                  >
                    <input
                      class="bp3-input"
                      data-testid="maxMemory-input"
                      type="text"
                      value="100"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="StyledProps--font StyledProps--main StyledProps--padding-left-large"
          >
            <div
              class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
            >
              <div
                class="Text--withDocsTooltip"
                data-tooltip-id="prefMinNodeCount"
              >
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-variation-small-semi"
                >
                  ce.nodeRecommendation.prefMinNodeCount
                </p>
              </div>
              <div
                class="Layout--horizontal Layout--layout-spacing-small StyledProps--main minNodeContainer"
              >
                <button
                  class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--withLeftIcon Button--variation Button--variation-secondary"
                  data-testid="increment-minnode-btn"
                  type="button"
                >
                  <span
                    class="bp3-icon bp3-icon-minus StyledProps--main"
                    icon="minus"
                  >
                    <svg
                      data-icon="minus"
                      height="16"
                      viewBox="0 0 16 16"
                      width="16"
                    >
                      <desc>
                        minus
                      </desc>
                      <path
                        d="M13 7H3c-.55 0-1 .45-1 1s.45 1 1 1h10c.55 0 1-.45 1-1s-.45-1-1-1z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </button>
                <div
                  class="TextInput--main minNodeInput"
                >
                  <div
                    class="bp3-input-group"
                  >
                    <input
                      class="bp3-input"
                      data-testid="minnodes-input"
                      type="text"
                      value="10"
                    />
                  </div>
                </div>
                <button
                  class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--withLeftIcon Button--variation Button--variation-secondary"
                  data-testid="decrement-minnode-btn"
                  type="button"
                >
                  <span
                    class="bp3-icon bp3-icon-plus StyledProps--main"
                    icon="plus"
                  >
                    <svg
                      data-icon="plus"
                      height="16"
                      viewBox="0 0 16 16"
                      width="16"
                    >
                      <desc>
                        plus
                      </desc>
                      <path
                        d="M13 7H9V3c0-.55-.45-1-1-1s-1 .45-1 1v4H3c-.55 0-1 .45-1 1s.45 1 1 1h4v4c0 .55.45 1 1 1s1-.45 1-1V9h4c.55 0 1-.45 1-1s-.45-1-1-1z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="StyledProps--font StyledProps--main"
      >
        <div
          class="StyledProps--font StyledProps--main StyledProps--margin-bottom-small"
        >
          <span
            class="StyledProps--font StyledProps--main StyledProps--inline StyledProps--font-variation-small-semi"
          >
            ce.nodeRecommendation.preferredInstanceFamilies
          </span>
        </div>
        <button
          class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--margin-bottom-small Button--with-current-color Button--iconOnly Button--withLeftIcon Button--variation Button--variation-link"
          type="button"
        >
          <span
            class="bp3-icon bp3-icon-plus StyledProps--main"
            icon="plus"
          >
            <svg
              data-icon="plus"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                plus
              </desc>
              <path
                d="M13 7H9V3c0-.55-.45-1-1-1s-1 .45-1 1v4H3c-.55 0-1 .45-1 1s.45 1 1 1h4v4c0 .55.45 1 1 1s1-.45 1-1V9h4c.55 0 1-.45 1-1s-.45-1-1-1z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <span
            class="bp3-button-text"
          >
            ce.nodeRecommendation.addPreferredInstanceFamilies
          </span>
        </button>
      </div>
      <div
        class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
      >
        <button
          class="bp3-button bp3-disabled bp3-loading Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--variation Button--variation-primary"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <div
            class="bp3-spinner bp3-button-spinner"
          >
            <div
              class="bp3-spinner-animation"
            >
              <svg
                height="20"
                stroke-width="16.00"
                viewBox="-3.00 -3.00 106.00 106.00"
                width="20"
              >
                <path
                  class="bp3-spinner-track"
                  d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                />
                <path
                  class="bp3-spinner-head"
                  d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                  pathLength="280"
                  stroke-dasharray="280 280"
                  stroke-dashoffset="210"
                />
              </svg>
            </div>
          </div>
          <span
            class="bp3-button-text"
          >
            ce.nodeRecommendation.applyPreferences
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`test cases for tune recommendation card should be able to render tune recommendation card hidden 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main spaceBetween StyledProps--padding-medium StyledProps--flex"
  >
    <p
      class="StyledProps--font StyledProps--main pointer StyledProps--font-variation-h6 StyledProps--color StyledProps--color-primary7"
    >
      ce.nodeRecommendation.setInstancePreferences
    </p>
    <span
      class="bp3-icon bp3-icon-caret-down StyledProps--main pointer StyledProps--color StyledProps--color-primary7"
      icon="caret-down"
    >
      <svg
        data-icon="caret-down"
        height="16"
        viewBox="0 0 16 16"
        width="16"
      >
        <desc>
          caret-down
        </desc>
        <path
          d="M12 6.5c0-.28-.22-.5-.5-.5h-7a.495.495 0 00-.37.83l3.5 4c.09.1.22.17.37.17s.28-.07.37-.17l3.5-4c.08-.09.13-.2.13-.33z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`test cases for tune recommendation card should be able to render tune recommendation card when there are instance families 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main spaceBetween StyledProps--padding-medium StyledProps--flex"
  >
    <p
      class="StyledProps--font StyledProps--main pointer StyledProps--font-variation-h6 StyledProps--color StyledProps--color-grey800"
    >
      ce.nodeRecommendation.setInstancePreferences
    </p>
    <span
      class="bp3-icon bp3-icon-caret-up StyledProps--main pointer StyledProps--color StyledProps--color-primary7"
      icon="caret-up"
    >
      <svg
        data-icon="caret-up"
        height="16"
        viewBox="0 0 16 16"
        width="16"
      >
        <desc>
          caret-up
        </desc>
        <path
          d="M11.87 9.17s.01 0 0 0l-3.5-4C8.28 5.07 8.15 5 8 5s-.28.07-.37.17l-3.5 4a.495.495 0 00.37.83h7a.495.495 0 00.37-.83z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
  </div>
  <div
    class="StyledProps--font StyledProps--main preferences StyledProps--padding-medium StyledProps--background-primary1"
  >
    <div
      class="Layout--vertical Layout--layout-spacing-medium StyledProps--main"
    >
      <div
        class="StyledProps--font StyledProps--main"
      >
        <div
          class="Text--withDocsTooltip"
          data-tooltip-id="prefResourceNeeds"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-small-semi"
          >
            ce.nodeRecommendation.prefResourceNeeds
          </p>
        </div>
        <div
          class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main spaceBetween StyledProps--flex StyledProps--margin-top-small"
        >
          <div
            class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
          >
            <div
              class="StyledProps--font StyledProps--main"
            >
              <div
                class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
              >
                <div
                  class="StyledProps--font StyledProps--main spaceBetween StyledProps--flex"
                >
                  <span
                    class="StyledProps--font StyledProps--main StyledProps--padding-right-small StyledProps--inline StyledProps--color StyledProps--color-grey500 StyledProps--font-variation-small-semi"
                  >
                    ce.nodeRecommendation.cpus
                  </span>
                  <div
                    class="TextInput--main input"
                  >
                    <div
                      class="bp3-input-group"
                    >
                      <input
                        class="bp3-input"
                        data-testid="sumCpu-input"
                        type="text"
                        value="30"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="StyledProps--font StyledProps--main spaceBetween StyledProps--flex"
                >
                  <span
                    class="StyledProps--font StyledProps--main StyledProps--padding-right-small StyledProps--inline StyledProps--color StyledProps--color-grey500 StyledProps--font-variation-small-semi"
                  >
                    ce.nodeRecommendation.mem
                  </span>
                  <div
                    class="TextInput--main input"
                  >
                    <div
                      class="bp3-input-group"
                    >
                      <input
                        class="bp3-input"
                        data-testid="sumMem-input"
                        type="text"
                        value="120"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-center"
            >
              <div
                class="StyledProps--font StyledProps--main"
                style="border-width: 1px; border-style: solid; border-left: 0; width: 14px; height: 60px;"
              />
            </div>
            <div
              class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-center StyledProps--margin-right-small"
            >
              <span
                class="bp3-icon bp3-icon-plus StyledProps--main StyledProps--color StyledProps--color-grey700"
                icon="plus"
              >
                <svg
                  data-icon="plus"
                  height="12"
                  viewBox="0 0 16 16"
                  width="12"
                >
                  <desc>
                    plus
                  </desc>
                  <path
                    d="M13 7H9V3c0-.55-.45-1-1-1s-1 .45-1 1v4H3c-.55 0-1 .45-1 1s.45 1 1 1h4v4c0 .55.45 1 1 1s1-.45 1-1V9h4c.55 0 1-.45 1-1s-.45-1-1-1z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </div>
          </div>
          <div
            class="StyledProps--font StyledProps--main"
            style="width: 60%;"
          >
            <div
              class="Layout--vertical StyledProps--main"
            >
              <div
                class="Layout--horizontal StyledProps--main"
                style="justify-content: space-between;"
              >
                <div
                  class="Text--withDocsTooltip"
                  data-tooltip-id="buffer"
                >
                  <p
                    class="StyledProps--font StyledProps--main StyledProps--font-variation-small-semi"
                    style="display: flex; align-items: center;"
                  >
                    ce.nodeRecommendation.buffer
                    <span
                      class="bp3-icon StyledProps--main Text--icon StyledProps--padding-left-xsmall StyledProps--color StyledProps--color-grey800"
                      data-icon="info"
                    >
                      <svg
                        height="12"
                        viewBox="0 0 12 12"
                        width="12"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M0 6a6 6 0 1012 0A6 6 0 000 6zm.75 0a5.25 5.25 0 1010.5 0A5.25 5.25 0 00.75 6zm4.206 3.104v.646h2.682v-.646l-.836-.144V4.203H4.956v.65l.836.144V8.96l-.836.144zm1.846-7.352v1.03h-1.01v-1.03h1.01z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                  </p>
                </div>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-variation-small-semi StyledProps--color StyledProps--color-grey400"
                >
                  0%
                </p>
              </div>
              <div
                class="bp3-slider bp3-slider-unlabeled bufferSlider"
              >
                <div
                  class="bp3-slider-track"
                >
                  <div
                    class="bp3-slider-progress bp3-intent-primary"
                    style="left: 0.00%; right: 100.00%; top: 0px;"
                  />
                  <div
                    class="bp3-slider-progress"
                    style="left: 0.00%; right: 100.00%; top: 0px;"
                  />
                  <div
                    class="bp3-slider-progress"
                    style="left: 0.00%; right: 0.00%; top: 0px;"
                  />
                </div>
                <div
                  class="bp3-slider-axis"
                />
                <span
                  class="bp3-slider-handle"
                  style="left: calc(0.00% - 0px);"
                  tabindex="0"
                />
              </div>
              <p
                class="StyledProps--font StyledProps--main StyledProps--font-variation-small StyledProps--margin-top-xsmall"
              >
                ce.nodeRecommendation.cpuAndMemeValWithBuffer
              </p>
            </div>
          </div>
        </div>
        <div
          class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main StyledProps--padding-top-medium"
        >
          <div
            class="StyledProps--font StyledProps--main"
          >
            <div
              class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
            >
              <div
                class="Text--withDocsTooltip"
                data-tooltip-id="largestWorkloadReq"
              >
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-variation-small-semi"
                >
                  ce.nodeRecommendation.largestWorkloadReq
                </p>
              </div>
              <div
                class="StyledProps--font StyledProps--main spaceBetween StyledProps--flex"
                style="width: 75%;"
              >
                <span
                  class="StyledProps--font StyledProps--main StyledProps--inline StyledProps--color StyledProps--color-grey500 StyledProps--font-variation-small-semi"
                >
                  ce.nodeRecommendation.cpus
                </span>
                <div
                  class="TextInput--main input"
                >
                  <div
                    class="bp3-input-group"
                  >
                    <input
                      class="bp3-input"
                      data-testid="maxCpu-input"
                      type="text"
                      value="44"
                    />
                  </div>
                </div>
              </div>
              <div
                class="StyledProps--font StyledProps--main spaceBetween StyledProps--flex"
                style="width: 75%;"
              >
                <span
                  class="StyledProps--font StyledProps--main StyledProps--inline StyledProps--color StyledProps--color-grey500 StyledProps--font-variation-small-semi"
                >
                  ce.nodeRecommendation.mem
                </span>
                <div
                  class="TextInput--main input"
                >
                  <div
                    class="bp3-input-group"
                  >
                    <input
                      class="bp3-input"
                      data-testid="maxMemory-input"
                      type="text"
                      value="100"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="StyledProps--font StyledProps--main StyledProps--padding-left-large"
          >
            <div
              class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
            >
              <div
                class="Text--withDocsTooltip"
                data-tooltip-id="prefMinNodeCount"
              >
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-variation-small-semi"
                >
                  ce.nodeRecommendation.prefMinNodeCount
                </p>
              </div>
              <div
                class="Layout--horizontal Layout--layout-spacing-small StyledProps--main minNodeContainer"
              >
                <button
                  class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--withLeftIcon Button--variation Button--variation-secondary"
                  data-testid="increment-minnode-btn"
                  type="button"
                >
                  <span
                    class="bp3-icon bp3-icon-minus StyledProps--main"
                    icon="minus"
                  >
                    <svg
                      data-icon="minus"
                      height="16"
                      viewBox="0 0 16 16"
                      width="16"
                    >
                      <desc>
                        minus
                      </desc>
                      <path
                        d="M13 7H3c-.55 0-1 .45-1 1s.45 1 1 1h10c.55 0 1-.45 1-1s-.45-1-1-1z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </button>
                <div
                  class="TextInput--main minNodeInput"
                >
                  <div
                    class="bp3-input-group"
                  >
                    <input
                      class="bp3-input"
                      data-testid="minnodes-input"
                      type="text"
                      value="10"
                    />
                  </div>
                </div>
                <button
                  class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--withLeftIcon Button--variation Button--variation-secondary"
                  data-testid="decrement-minnode-btn"
                  type="button"
                >
                  <span
                    class="bp3-icon bp3-icon-plus StyledProps--main"
                    icon="plus"
                  >
                    <svg
                      data-icon="plus"
                      height="16"
                      viewBox="0 0 16 16"
                      width="16"
                    >
                      <desc>
                        plus
                      </desc>
                      <path
                        d="M13 7H9V3c0-.55-.45-1-1-1s-1 .45-1 1v4H3c-.55 0-1 .45-1 1s.45 1 1 1h4v4c0 .55.45 1 1 1s1-.45 1-1V9h4c.55 0 1-.45 1-1s-.45-1-1-1z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="StyledProps--font StyledProps--main"
      >
        <div
          class="StyledProps--font StyledProps--main StyledProps--margin-bottom-small"
        >
          <span
            class="StyledProps--font StyledProps--main StyledProps--inline StyledProps--font-variation-small-semi"
          >
            ce.nodeRecommendation.preferredInstanceFamilies
          </span>
          <button
            class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--inline Button--with-current-color Button--iconOnly Button--withLeftIcon Button--variation Button--variation-link"
            type="button"
          >
            <span
              class="bp3-icon bp3-icon-edit StyledProps--main"
              icon="edit"
            >
              <svg
                data-icon="edit"
                height="12"
                viewBox="0 0 16 16"
                width="12"
              >
                <desc>
                  edit
                </desc>
                <path
                  d="M3.25 10.26l2.47 2.47 6.69-6.69-2.46-2.48-6.7 6.7zM.99 14.99l3.86-1.39-2.46-2.44-1.4 3.83zm12.25-14c-.48 0-.92.2-1.24.51l-1.44 1.44 2.47 2.47 1.44-1.44c.32-.32.51-.75.51-1.24.01-.95-.77-1.74-1.74-1.74z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
            <span
              class="bp3-button-text"
            >
              edit
            </span>
          </button>
        </div>
        <div
          class="TextInput--main"
        >
          <div
            class="bp3-input-group instaceFamilyInput"
          >
            <input
              class="bp3-input"
              contenteditable="false"
              readonly=""
              type="text"
              value="type1"
            />
          </div>
        </div>
        <button
          class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--margin-bottom-small Button--with-current-color Button--iconOnly Button--withLeftIcon Button--variation Button--variation-link"
          type="button"
        >
          <span
            class="bp3-icon bp3-icon-plus StyledProps--main"
            icon="plus"
          >
            <svg
              data-icon="plus"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                plus
              </desc>
              <path
                d="M13 7H9V3c0-.55-.45-1-1-1s-1 .45-1 1v4H3c-.55 0-1 .45-1 1s.45 1 1 1h4v4c0 .55.45 1 1 1s1-.45 1-1V9h4c.55 0 1-.45 1-1s-.45-1-1-1z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <span
            class="bp3-button-text"
          >
            ce.nodeRecommendation.addPreferredInstanceFamilies
          </span>
        </button>
      </div>
      <div
        class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
      >
        <button
          class="bp3-button bp3-disabled bp3-loading Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--variation Button--variation-primary"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <div
            class="bp3-spinner bp3-button-spinner"
          >
            <div
              class="bp3-spinner-animation"
            >
              <svg
                height="20"
                stroke-width="16.00"
                viewBox="-3.00 -3.00 106.00 106.00"
                width="20"
              >
                <path
                  class="bp3-spinner-track"
                  d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                />
                <path
                  class="bp3-spinner-head"
                  d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                  pathLength="280"
                  stroke-dasharray="280 280"
                  stroke-dashoffset="210"
                />
              </svg>
            </div>
          </div>
          <span
            class="bp3-button-text"
          >
            ce.nodeRecommendation.applyPreferences
          </span>
        </button>
        <button
          class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--variation Button--variation-secondary"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            ce.recommendation.detailsPage.resetRecommendationText
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
